test/unit/math/test_ad_matvar_test.cpp

```diff
@@ -0,0 +1,371 @@
+#include <test/unit/math/test_ad.hpp>
+#include <gtest/gtest.h>
+#include <gtest/gtest-spi.h>
+#include <limits>
+#include <type_traits>
+#include <vector>
+
+template <typename T>
+auto one_arg_bad_vals(const T& x) {
+  return x;
+}
+
+template <typename T>
+auto one_arg_bad_vals(const stan::math::var_value<T>& x) {
+  return stan::math::make_callback_var(
+      x.val() * 0, [x](const auto& vi) mutable { x.adj() += vi.adj(); });
+}
```

Comment:
I'm confused about what this is testing?

Reply:
This is testing that functions with badly implemented autodiff fail when they should fail.

It's easy to accidentally make a bug in these tests so that everything passes :D.

For instance the first time I coded this I wasn't checking that values with varmat matched matvar.